### PR TITLE
feat(esl-utils): event utils destructuring

### PR DESCRIPTION
**Pull request by ala-n**
BREAKING CHANGE:
 - `isMouseEvent`, `isTouchEvent` moved outside of `EventUtils`
 - `normalizeTouchPoint` renamed to `touchPoint` and moved outside of `EventUtils`
 - `normalizeCoordinates` removed `offsetPoint` introduced instead
 to be used in combination with `touchPoint`

### CODE REVIEW

File: src/modules/esl-utils/dom/events/utils.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {ExportNs} from '../../environment/export-ns';
+
+@ExportNs('EventUtils')
+export class EventUtils {
+  /**
+   * Dispatch custom event.
```
Comment (NastaLeo) [inline src/modules/esl-utils/dom/events/utils.ts:6]: Dispatches custom event.

File: src/modules/esl-utils/dom/events/misc.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/** Check if the passed event is {@link MouseEvent} */
```
Comment (NastaLeo) [inline src/modules/esl-utils/dom/events/misc.ts:1]: Checks if the passed event is {@link MouseEvent}

File: src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts [hunk 3]
```diff
@@ -101,15 +101,15 @@ export class ESLToggleableDispatcher extends ESLBaseElement {
   /** Hide active element before e.target will be shown */
   @bind
   protected _onBeforeShow(e: CustomEvent): void {
-    const target = EventUtils.source(e);
+    const target = getCompositeTarget(e);
     if (!this.isAcceptable(target)) return;
     this.hideActive(target.groupName, target);
   }
 
   /** Update active element after a new element is shown */
```
Comment (NastaLeo) [inline src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts:109]: Updates active element after a new element is shown

File: src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts [hunk 4]
```diff
@@ -118,7 +118,7 @@ export class ESLToggleableDispatcher extends ESLBaseElement {
   /** Update group state after active element is hidden */
```
Comment (NastaLeo) [inline src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts:118]: Updates group state after active element is hidden

File: src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts [hunk 5]
```diff
@@ -127,7 +127,7 @@ export class ESLToggleableDispatcher extends ESLBaseElement {
   /** Update active elements */
```
Comment (NastaLeo) [inline src/modules/esl-toggleable/core/esl-toggleable-dispatcher.ts:127]: Updates active elements

File: src/modules/esl-scrollbar/core/esl-scrollbar.ts [hunk 6]
```diff
@@ -267,8 +268,8 @@ export class ESLScrollbar extends ESLBaseElement {
     }
 
     // Subscribe inverse handlers
```
Comment (NastaLeo) [inline src/modules/esl-scrollbar/core/esl-scrollbar.ts:270]: Subscribes inverse handlers

File: src/modules/esl-scrollbar/core/esl-scrollbar.ts [hunk 7]
```diff
@@ -294,13 +295,13 @@ export class ESLScrollbar extends ESLBaseElement {
     this.$target?.style.setProperty('scroll-behavior', 'auto');
     // Attach drag listeners
```
Comment (NastaLeo) [inline src/modules/esl-scrollbar/core/esl-scrollbar.ts:296]: Attaches drag listeners

**Review by NastaLeo — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

**Pull request merged**